pkg/apis/core/types.go

```diff
@@ -2568,9 +2568,10 @@ type PodSpec struct {
 	// file if specified. This is only valid for non-hostNetwork pods.
 	// +optional
 	HostAliases []HostAlias
-	// If specified, indicates the pod's priority. "SYSTEM" is a special keyword
-	// which indicates the highest priority. Any other name must be defined by
-	// creating a PriorityClass object with that name.
+	// If specified, indicates the pod's priority. "system-node-critical" and
```

Comment:
@bsalamat - Out of curiosity, why  does `system-node-critical` has higher priority than `system-cluster-critical`? Why isn't it otherway around?

Reply:
> Out of curiosity, why system-node-critical has higher priority than system-cluster-critical? Why isn't it otherway around?

`system-node-critical` is used for Pods that are critical and must run on the current node. `system-cluster-critical` are used for Pods that are critical to run in the cluster, but they don't need to run on the current node and things should be fine as long as they run in the cluster. That's why `system-node-critical` has higher priority. If at some point, we reach to a point that we have to choose to preempt Pods having `system-cluster-critical` and `system-node-critical`, we choose the former, because those Pods can be recreated and run on a different node.

Reply:
Thanks @bsalamat  - So, in a way, are `system-node-critical` pods static pods with critical pod annotation. I ask because as per https://github.com/kubernetes/kubernetes/issues/58379#issuecomment-359625527, it seems we are treating static pods no differently when compared to normal pods while evicting at kubelet level.

Reply:
I won't call pods with `system-node-critical` priority static pods. This is merely the priority of Pods. Pods with this priority can be scheduled by the scheduler and may not have a preassigned node.